my-app/app/components/add-exp.js

```diff
@@ -0,0 +1,158 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+    expenses: [],
+    users: [],
+    balances: [],
+    showContent: [],
+    limit: 10,
+    newExpense: false,
+    splitUnequally: false,
+    success: false,
+    expensesInit: function() {
+        var i = 0;
+        let expenses = this.get("expenses");
+        let limit = this.get("limit");
+        let showContent = this.get("showContent");
+        for(i = (expenses.length) - 1; i >= (expenses.length) - limit; i--){
+            if(i >= 0){
+                showContent.pushObject(expenses[i]);
+            }
+        }
+    }.on('init'),
+    actions: {
+        addExpense: function () {
+            var i = 0;
+            var expense = {};
+            var unequalList = [];
+            var temp = "";
+            expense.amount = this.get("amount");
+            expense.description = this.get("description");
+            var d = new Date();
+            expense.date = d.getFullYear() + '-' + (d.getMonth() + 1) + '-' + d.getDate();
+            let users = this.get("users");
+            users.forEach(function (item) {
+                if (item.hasPaid) {
+                    expense.from_id = item.name;
+                }
+            });
+            if (this.get('splitUnequally')) {
+                for (i = 0; i < users.length; i++) {
+                    if (users[i].hasReceived) {
+                        expense.to_id = users[i].name;
+                        expense.amount = users[i].unequalAmount;
+                        if (expense.to_id !== expense.from_id) {
+                            temp = JSON.stringify(expense);
+                            unequalList.pushObject(temp);
+                        }
+                    }
+                }
+                this.sendAction('action', unequalList);
+            }
+            else {
+                for (i = 0; i < users.length; i++) {
+                    if (users[i].hasReceived) {
+                        if (expense.to_id) {
+                            expense.to_id = expense.to_id + ",";
+                            expense.to_id = expense.to_id + users[i].name;
+                        }
+                        else {
+                            expense.to_id = users[i].name;
+                        }
+                    }
+                }
+                temp = JSON.stringify(expense);
+                this.sendAction('action', temp);
+            }
+            this.set('success', true);
+            Ember.run.later((function () {
```

Comment:
This is incorrect. What if your backend takes more than 1500 ms to store and respond in some case. Then might be reloading the page before the data is saved. 
Calling `window.location.reload` is also an incorrect way. You just want reload the data instead you are reloading the whole ember app. This app is quite small but at enterprise level it will just kill all advantages of having a client side app. 

Just do `model.reload()` in your route's action, that's enough.